OpenRA.Mods.Common/Traits/Upgrades/DeployToUpgrade.cs

```diff
@@ -89,59 +100,27 @@ public Order IssueOrder(Actor self, IOrderTargeter order, Target target, bool qu
 
 		public void ResolveOrder(Actor self, Order order)
 		{
-			if (order.OrderString != "DeployToUpgrade")
+			if (order.OrderString != "DeployToUpgrade" || deployState == DeployState.Deploying)
 				return;
 
 			if (!IsOnValidTerrain())
```

Comment:
Eh, it turns out that neither `TargetLocation` nor `ExtraLocation` are set in the order (which kinda makes sense, considering that you can just (un)deploy using a hotkey). I propose this change instead:

```diff
diff --git a/OpenRA.Mods.Common/Traits/Upgrades/DeployToUpgrade.cs b/OpenRA.Mods.Common/Traits/Upgrades/DeployToUpgrade.cs
index 523e879..84a2b53 100644
--- a/OpenRA.Mods.Common/Traits/Upgrades/DeployToUpgrade.cs
+++ b/OpenRA.Mods.Common/Traits/Upgrades/DeployToUpgrade.cs
@@ -103,9 +103,6 @@ public void ResolveOrder(Actor self, Order order)
                        if (order.OrderString != "DeployToUpgrade" || deployState == DeployState.Deploying)
                                return;
 
-                       if (!IsOnValidTerrain())
-                               return;
-
                        if (!order.Queued)
                                self.CancelActivity();
 
@@ -167,6 +164,9 @@ void Deploy()
                        if (deployState != DeployState.Undeployed)
                                return;
 
+                       if (!IsOnValidTerrain())
+                               return;
+
                        if (!string.IsNullOrEmpty(info.DeploySound))
                                Game.Sound.Play(info.DeploySound, self.CenterPosition);
```